lib/dictionary_processing.py

```diff
@@ -135,27 +142,68 @@ def add_service_urls_using_metamodel(
                 else:
                     package_dict.setdefault(package, [service_path])
             else:
-                print("Service doesnot belong to either /api or /rest ", service)
+                print("Service does not belong to either /api or /rest ", service)
+        if service_type == ServiceType.MIXED and not blacklist_utils.isBlacklistedForRest(service):
+            service_url = rest_services.get(service, None)
+            if service_url is not None:
+                service_path = get_service_path_from_service_url(
+                    service_url, rest_navigation_handler.get_rest_navigation_url())
+                service_urls_map[service_path] = (service, '/mixed')
+                package = service_path.split('/')[3]
+                if package in package_dict_deprecated:
+                    packages = package_dict_deprecated[package]
+                    packages.append(service_path)
+                else:
+                    package_dict_deprecated.setdefault(package, [service_path])
+            else:
+                print("Service does not belong to either /api or /rest ", service)
+    if mixed:
+        return package_dict_api, package_dict, package_dict_deprecated, replacement_map
+
     return package_dict_api, package_dict
 
 
-def get_paths_inside_metamodel(service, service_dict):
+
+def get_paths_inside_metamodel(service, service_dict, mixed=False, replacement_map={}, service_url=None, rest_navigation_handler=None):
     path_list = set()
+    is_mixed = False
     for operation_id in service_dict[service].operations.keys():
         for request in service_dict[service].operations[operation_id].metadata.keys(
         ):
             if request.lower() in ('post', 'put', 'patch', 'get', 'delete'):
-                path_list.add(
-                    service_dict[service].operations[operation_id].metadata[request].elements['path'].string_value)
+                path = service_dict[service].operations[operation_id].metadata[request].elements['path'].string_value
+                path_list.add(path)
+
+                # Check whether the service contains both @RequestMapping and @Verb annotations
+                if mixed and 'RequestMapping' in service_dict[service].operations[operation_id].metadata.keys():
+                    is_mixed = True
```

Comment:
Having the 'mixed' parameter and the 'is_mixed' local variable, makes this code hard to read and somewhat controversial - mixed is both input and output here. Would some better naming be possible?

Reply:
This function is a mess, which I tidied up (a bit) and it is still a mess. It is doing multiple things - checking for RequestMapping, calling the Rest navigation service, looking for VERB etc. In my opinion, another processing class, should be responsible for distributing paths to corresponding /api, /rest or /deprecated packages, involving several strategies for handling different paths. 

I have deferred the full refactoring, since it requires a lot of changes to the method calling this one and the tests created.

Reply:
Sounds good, I just wondered if some trivial renaming of local vars can improve the readability